modules/new_serverpod_auth/serverpod_auth_idp/serverpod_auth_idp_server/test/passkey/passkey_accounts_test.dart

```diff
@@ -0,0 +1,173 @@
+import 'dart:convert';
+
+import 'package:serverpod/serverpod.dart';
+import 'package:serverpod_auth_idp_server/core.dart';
+import 'package:serverpod_auth_idp_server/providers/passkey.dart';
+import 'package:serverpod_auth_idp_server/serverpod_auth_idp_server.dart';
+import 'package:serverpod_auth_idp_server/src/providers/email/util/uint8list_extension.dart';
+import 'package:test/test.dart';
+
+import '../test_tools/serverpod_test_tools.dart';
+
+void main() {
+  withServerpod(
+    'Given a session for an existing `AuthUser` without authentication,',
+    (final sessionBuilder, final _) {
+      late AuthUserModel user;
+      late Session session;
+
+      final challengeId = const Uuid().v4obj();
+
+      setUp(() async {
+        user = await AuthUsers.create(sessionBuilder.build());
+
+        session = sessionBuilder
+            .copyWith(
+              authentication: AuthenticationOverride.authenticationInfo(
+                user.id.uuid,
+                {},
+              ),
+            )
+            .build();
+
+        await PasskeyChallenge.db.insertRow(
+          session,
+          PasskeyChallenge(
+            id: challengeId,
+            challenge: _registrationChallenge,
+          ),
+        );
+
+        PasskeyAccounts.config = PasskeyAccountConfig(
+          hostname: 'localhost',
+        );
+      });
+
+      test(
+          "when calling `PasskeyAccounts.registerPasskey`, then the passkey is registered for the session's user.",
```

Comment:
We should also test that the user can not complete the challenge after it has expired. Then, this one could be more specific like:

```suggestion
          "when calling `PasskeyAccounts.registerPasskey` before challenge expires, then the passkey is registered for the session's user.",
```